packages/admin-next/dashboard/src/v2-routes/promotions/promotion-list/components/promotion-list-table/promotion-list-table.tsx

```diff
@@ -0,0 +1,149 @@
+import { PencilSquare, Trash } from "@medusajs/icons"
+import { PromotionDTO } from "@medusajs/types"
+import { Button, Container, Heading, usePrompt } from "@medusajs/ui"
+import { createColumnHelper } from "@tanstack/react-table"
+import { useAdminDeleteDiscount } from "medusa-react"
+import { useMemo } from "react"
+import { useTranslation } from "react-i18next"
+import { Link, Outlet, useLoaderData } from "react-router-dom"
+import { useV2Promotions } from "../../../../../lib/api-v2"
+
+import { ActionMenu } from "../../../../../components/common/action-menu"
+import { DataTable } from "../../../../../components/table/data-table"
+import { usePromotionTableColumns } from "../../../../../hooks/table/columns-v2/use-promotion-table-columns"
+import { usePromotionTableFilters } from "../../../../../hooks/table/filters-v2/use-promotion-table-filters"
+import { usePromotionTableQuery } from "../../../../../hooks/table/query-v2/use-promotion-table-query"
+import { useDataTable } from "../../../../../hooks/use-data-table"
+import { promotionsLoader } from "../../loader"
+
+const PAGE_SIZE = 20
+
+export const PromotionListTable = () => {
+  const { t } = useTranslation()
+
+  // Question: what is this?
+  const initialData = useLoaderData() as Awaited<
+    ReturnType<ReturnType<typeof promotionsLoader>>
+  >
+
+  const { searchParams, raw } = usePromotionTableQuery({ pageSize: PAGE_SIZE })
+  const { promotions, count, isLoading, isError, error } = useV2Promotions(
+    { ...searchParams },
+    {
+      initialData,
+      // Question: what is this?
+      // TODO: reenable this once its figured out
```

Comment:
I believe this tells `react-query` to not delete old data when you fetch a new batch. For example, if you are doing infinite scrolling, the second request will contain additional 20 items, but we still want to keep the old 20 in memory.

Reply:
ah, that makes sense. Thanks!